src/AutoRest.CSharp/Output/Models/Types/OutputLibrary.cs

```diff
@@ -235,6 +260,14 @@ public TypeProvider FindTypeForSchema(Schema schema)
             _ => throw new NotImplementedException()
         };
 
+        private TypeProvider BuildResourceModel(Schema schema) => schema switch
+        {
+            SealedChoiceSchema sealedChoiceSchema => (TypeProvider)new EnumType(sealedChoiceSchema, _context),
```

Comment:
I think we should fail if its not an ObjectSchema here

Reply:
That's true. But till now we just see ObjectType, should we update the swagger to test other schema types?

Reply:
By failing we are asserting what we expect, if we ever see the failure happen we can adjust our expectations accordingly